scRNA-seq/post-processing_scRNA-seq.Rmd

```diff
@@ -0,0 +1,167 @@
+  # Obtain paths to data
+  barcode_loc <- file.path(base_path, "alevin", "quants_mat_rows.txt")
+  gene_loc <- file.path(base_path, "alevin", "quants_mat_cols.txt")
+  data_loc <- file.path(base_path, "alevin", "quants_mat.csv")
+    
+  if (!file.exists(barcode_loc)) {
+    stop("Barcode file missing")
+  }
+  if (!file.exists(gene_loc)) {
+    stop("Gene name file missing")
+  }
+  if (!file.exists(data_loc)) {
+    stop("Expression matrix file missing")
+  }
+  # Read in the data from Alevin output
+  expression_matrix <- readr::read_csv(data_loc, col_names = FALSE, 
+                                       progress = FALSE)
+  
+  # Transpose the matrix so it is gene x cell
+  expression_matrix <- t(expression_matrix[,1:ncol(expression_matrix)-1])
+  
+  # Apply the colnames and rownames to the dataset
+  colnames(expression_matrix) <- readLines(barcode_loc)
+  rownames(expression_matrix) <- readLines(gene_loc)
+    
+  # Make NA values into 0's
+  expression_matrix[is.na(expression_matrix)] <- 0
+  return(expression_matrix)
+}
+```
+
+Put the path to the Alevin output here:
+```{r Path to alevin output}
+# Store the path to the Alevin output files
+alevin_file <- file.path("alevin_output")
+```
+
+#### Perform QC checks with 'csoneson/alevinQC'
+This will provide html output with graphs evaluating the data. Keep in mind that 
+the data we have processed in this workshop is not the full dataset, and won't 
+look as good as the full set.
+```{r alevinQC}
+# Produce a QC report
+alevinQC::alevinQCReport(alevin_file,
+                         sampleId = "pbmc_1k_v2_S1_L001_subset", 
+                         outputFile = "pbmc_1k_v2_S1_L001_qc_report.html", 
+                         outputDir = "data",
+                         outputFormat = "html_document")
+```
+
+#### Import counts from Alevin output
+Here we are using the function we set up in the earlier chunk. 
+```{r Import Data}
+# Read in the data
+alv_data <- ReadAlevin(alevin_file)
+```
+
+#### Set up the Seurat object and apply a filter to the data. 
+In this example, we are applying a gene filter cutoff of at least 3 cells 
+expressing each gene. 
+For cell filtering, each cell must express at least 200 genes.
+```{r Set Up Seurat Object}
+# Set up the seurat object 
+seurat_obj <- CreateSeuratObject(raw.data = alv_data, min.cells = 3,
+                                 min.genes = 200, 
+                                 project = "1k_pbmc_subset")
+```
+
+#### Normalize the data using log normalization
+These are the default settings for normalization and should work in most 
+instances. 
+```{r Normalize the Data}
+seurat_norm <- NormalizeData(object = seurat_obj, 
+                             normalization.method = "LogNormalize",
+                             scale.factor = 10000)
+```
+
+#### Identify variable genes and scale the data
+In this step, we will identify genes with higher variance. 
+These are probably the genes that are of more interest to us and they are what 
```

Comment:
This is perhaps a note for when #27 is addressed, but 
> These are probably the genes that are of more interest to us

We should say a bit about the _why_ here